src/response/error.rs

```diff
@@ -1,70 +1,142 @@
+use std::{error::Error, fmt::Debug};
+
 use tonic::codegen::http;
 
+use crate::auth::AuthError;
+
 /// Exception type for resulting from invalid interactions with Momento Services.
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 pub enum MomentoError {
     /// Momento Service encountered an unexpected exception while trying to fulfill the request.
-    InternalServerError(String),
+    #[error("internal server error: {description}")]
+    InternalServerError {
+        description: String,
```

Comment:
As a suggestion, use `Cow<'static, str>` here and in the other enum variants so that you can avoid allocating a string in most of the error cases.

Reply:
sure, nbd